server/controllers/user.go

```diff
@@ -1,59 +1,29 @@
 package controllers
 
 import (
-	"context"
-	"fmt"
-	"log"
-	"net/http"
 	"server/configs"
-
-	"github.com/gin-gonic/gin"
-	// "go.mongodb.org/mongo-driver/bson"
+	"server/models"
+	"server/services"
 )
 
-func CreateUserHandler(ctx *gin.Context) {
-	// var user User
-
-	// err := json.NewDecoder(r.Body).Decode(&user)
-	// if err != nil {
-	// 	http.Error(w, "Bad request", http.StatusBadRequest)
-	// 	return
-	// }
-
-	err := createUser(User{
-		Name:  "Jean",
-		Email: "jean@epitech.eu",
-	})
-
-	if err != nil {
-		fmt.Print("HAHAHAHHA")
-		// http.Error(w, "Failed to add user", http.StatusInternalServerError)
-		return
-	}
+var CreateUserService = services.CreateUser
+var FindOneUserService = services.FindUser
 
-	ctx.JSON(http.StatusCreated, Response{Message: "user created"})
-}
+// CreateUserHandler handles the creation of a new user.
+func CreateUserHandler(user models.User) {
+	// Get the user collection from the configuration
+	userCollection, _ := configs.GetCollection("users")
 
-func createUser(user User) error {
-	collection, _ := configs.GetCollection("users")
+	// Check if the user already exists in the database
+	result := FindOneUserService(userCollection, user.NodeID)
 
-	// Insertion d'un document
-	insertResult, err := collection.InsertOne(context.TODO(), user)
-	if err != nil {
-		log.Fatalf("Erreur lors de l'insertion du document : %v", err)
+	// If the user doesn't exist, create a new user
+	if result == nil {
+		CreateUserService(userCollection, user)
 	}
-
-	// Affichage de l'ID du document inséré
-	fmt.Printf("Document inséré avec l'ID : %v\n", insertResult.InsertedID)
-
-	return nil
-}
-
-type User struct {
-	Name  string `bson:"name"`
-	Email string `bson:"email"`
 }
 
+// Response represents the JSON response for HTTP requests.
 type Response struct {
```

Comment:
Move this type to models or delete it. It may be useless now.
We also don't need ping controller anymore